woocommerce/woocommerce-ios
# Coupon Editing: random coupon generation, free shipping toggle and coupon restrictions view model

Part of #6488 

### Description
In this PR:
- Implemented the generation of the random coupon code.
- Implemented the toggle for the free shipping.
- Handled correctly the coupon restrictions view model.

Added @toupper between the reviewers because of his onboarding process.

### Testing instructions
1. Open a coupon detail and make sure the feature flag for editing a coupon is enabled.
2. Press the ellipsis menu in the navigation menu.
3. Try to edit the toggle for the free shipping.
4. Try to press the button for generating a new coupon code randomly. You should see a new coupon code with 8 characters.
5. Open the Usage Restrictions option, you should see like before the view for editing the restrictions.


---
